### PR TITLE
code cleanup: using <> where possible

### PR DESCRIPTION
**Pull request by lbergelson**
replacing redundantly specified explicit types with `<>` where possible using intellij autofixing magic

### CODE REVIEW

**Comment by lbergelson** [comment]
@droazen Here's a simple one for you.  Gets ride of all those yellow marks in intellij that drive me crazy.

File: src/main/java/htsjdk/samtools/util/IntervalTree.java [hunk 1]
```diff
@@ -74,7 +74,7 @@ public V put( final int start, final int end, final V value )
 
         if ( mRoot == null )
         {
-            mRoot = new Node<V>(start,end,value);
+            mRoot = new Node<>(start, end, value);
```
Comment (yfarjoun) [inline src/main/java/htsjdk/samtools/util/IntervalTree.java]: you said "only <T> -> <>"...here there are spaces too! 😄

File: src/test/java/htsjdk/samtools/util/IntervalTreeMapTest.java [hunk 2]
```diff
@@ -31,7 +31,7 @@
 public class IntervalTreeMapTest {
     @Test
     public void testBasic() {
-        IntervalTreeMap<Interval> m=new IntervalTreeMap<Interval>();
+        IntervalTreeMap<Interval> m= new IntervalTreeMap<>();
```
Comment (yfarjoun) [inline src/test/java/htsjdk/samtools/util/IntervalTreeMapTest.java]: either spaces on both sides, or don't change it...

**Comment by yfarjoun** [comment]
👍 fix conflicts and the strange one-sided space change and we're good!

**Comment by lbergelson** [comment]
closing this one.

**Pull request closed without merging**